### PR TITLE
pass endpoint to url builder

### PR DESCRIPTION
**Pull request by jaredkosanovic**

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy [hunk 1]
```diff
@@ -136,16 +136,18 @@ abstract class Resource {
      * The path is copied from the request path.
      * The query parameters are taken from the params argument.
      *
-     * Falsey params are omitted from the url.
+     * Falsely params are omitted from the url.
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy]: Falsey is correct

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy [hunk 2]
```diff
@@ -136,16 +136,18 @@ abstract class Resource {
...
      * converted to page[number] and page[size].
      *
      * @param params a map of query parameters for the url
+     * @param resourceEndpoint: the endpoint to be appended on the uri.
      * @return the url
      */
-    protected String getPaginationUrl(Map params) {
-        URIBuilder uriBuilder = new URIBuilder(endpointUri).setPath(uriInfo.requestUri.path)
+    protected String getPaginationUrl(Map params, String resourceEndpoint) {
+        String path = endpointUri.path + resourceEndpoint
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy]: Is there a method you can use to join paths? Let's not assume that endpointUri.path ends with a slash, or that resourceEndpoint starts with one. Or if we are going to assume that, at least document it somewhere.

**Review by magical — CHANGES_REQUESTED** [review]

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy (hunk 1 above)
Comment (jaredkosanovic) [reply src/main/groovy/edu/oregonstate/mist/api/Resource.groovy]: Ah yes. I've seen it as falsy too. I guess I should read a whole sentence before correcting a word.

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy (hunk 2 above)
Comment (floppyears) [reply src/main/groovy/edu/oregonstate/mist/api/Resource.groovy]: good point

File: src/main/groovy/edu/oregonstate/mist/api/Resource.groovy [hunk 3]
```diff
@@ -145,9 +147,8 @@ abstract class Resource {
      * @return the url
      */
     protected String getPaginationUrl(Map params, String resourceEndpoint) {
-        String path = endpointUri.path + resourceEndpoint
-
-        URIBuilder uriBuilder = new URIBuilder(endpointUri).setPath(path)
+        URI baseUri = UriBuilder.fromUri(endpointUri).path(resourceEndpoint).build()
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/api/Resource.groovy]: UriBuilder preserves the original path of the uri so I'm using that to build a uri, then basing the URIBuilder object off the path of baseUri.

**Pull request merged**